### ISSUE TITLE
object integer types are not handled correctly in detect_types

### ISSUE THREAD

**Issue by amueller** [dabl/dabl#69]
If an integer column is stored as object type, it's becoming continuous because it's handled as "dirty float". That requires some fixes.
Maybe that will also allow us to fix handing low cardinality floats as categorical.

**Comment by Motorrat** [comment]
I have a question: if the integers are to be handled as numbers, what's the problem of having them converted into float? If they are to be handled as categorical, how do we know that? 
 
I was also confronted with the similar problem and have decided to treat everything that can be coerced into float as such, the rest is treated as categorical.
https://github.com/paypal/autosklearn-zeroconf/blob/master/bin/zeroconf.py#L136

```
for col in dataframe.select_dtypes(exclude=[np.number]).columns:
        dataframe[col] = dataframe[col].astype('category').cat.codes
```

**Comment by amueller** [comment]
>  If they are to be handled as categorical, how do we know that?

Right now based on a dirty, dirty hack:
https://github.com/amueller/dabl/blob/master/dabl/preprocessing.py#L60

if the histogram is "smooth enough" I assume it's continuous, otherwise it's categorical. The heuristic is pending a principled evaluation, though it's a bit hard to define ground truth.

**Comment by amueller** [comment]
Fixed by #270.
